src/app/[lng]/(infos)/profile/[walletAddress]/page.tsx

```diff
@@ -0,0 +1,32 @@
+import ProfilePage from '@/app/ui/profile/ProfilePage';
+import type { Metadata } from 'next';
+import { getSiteUrl } from '@/const/urls';
+
+type Props = {
+  params: { walletAddress: string };
+};
+
+const baseUrl = getSiteUrl();
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  return {
+    openGraph: {
+      title: `Profile of ${params.walletAddress}`,
+      description: `Profile of ${params.walletAddress}`,
+      type: 'profile',
+      url: `${baseUrl}/profile/${params.walletAddress}`,
+      images: `${baseUrl}/api/profile/${params.walletAddress}`,
+    },
+    twitter: {
+      images: `${baseUrl}/api/profile/${params.walletAddress}`,
+    },
+  };
+}
+
+export default async function Page({
+  params,
+}: {
+  params: { walletAddress: string };
+}) {
+  return <ProfilePage walletAddress={params.walletAddress} isPublic />;
```

Comment:
why not params?.walletAddress?

Reply:
this page is rendered only if params.walletAddress exists in the path (it is not optional in the url)